WordPress/gutenberg
# [Block Library]Rename Query and QueryLoop blocks to QueryLoop and PostTemplate respectively

## Description
Part of: https://github.com/WordPress/gutenberg/issues/30910

>Let's consider changing the names from "Query" and "Loop" to "Query Loop" and "Post Template" for more clarity.

This PR rename `Query` and `QueryLoop` blocks to `QueryLoop` and `PostTemplate` respectively.

## Notes
1. These two blocks will be shipped with 5.8 for the first time and if we decide on going through with this, these changes will need to be back ported asap.
2. By making this change it means that every block theme will need to update with the new blocks as they existing ones won't work as expected (invalid blocks).  
3. I have updated the block patterns to use `QueryLoop` and `unregister` the core `Query` patterns. If this lands we'll need to make the pattern changes and back port to core as well.
4. I had created a deprecated version of `Query` block but this is not needed right now as the block has been renamed, so I have deleted it in this PR.
5. Failing tests are mostly(haven't checked all them yet) about FSE tests that use a `tt1-blocks` version with the old block. This produces block invalidations and therefore `console.error` logs that make the tests fail. I'll look into this more but this must be done in coordination with the update of `tt1-block` theme and then update the version that GB tests are using. I have created a draft PR in `theme-experiments` repo here: https://github.com/WordPress/theme-experiments/pull/265

**Please test carefully and if you have some time it would be great to check about any missing strings that might be used for example in css class names or such details.** I think I've covered it all but lots of moving parts with renaming. Also be vigilant with the fact that the `Query` block has been renamed to `QueryLoop` so it might be a bit confusing to see the changes as `QueryLoop` was an existing block.


## Testing instructions
1. Insert `Query` block in a page/FSE and make changes in editor and view front-end as well.
2. Everything should work just as before.
3. Make sure the patterns work and the styles are the expected ones as before.
